settings/src/utilities.js

```diff
@@ -18,6 +18,12 @@ export function useGetUserRecord( userId ) {
 		userRecord.record.password = '';
 	}
 
+	const availableProviders = userRecord.record?.[ '2fa_available_providers' ] ?? [];
+	const primaryProviders = [ 'Two_Factor_Totp', 'TwoFactor_Provider_WebAuthn' ];
+	userRecord.hasPrimaryProvider = !! availableProviders.filter( ( provider ) =>
+		primaryProviders.includes( provider )
+	).length;
+
```

Comment:
🤔 This feels a bit unexpected to me, since it's mixing custom app data with a native Gutenberg object. `isSaving` is kind of doing that too, but it's more of a generic helper that Gutenberg will hopefully add one day.

A separate utility function like `hasPrimaryProvider( availableProviders ) {}` feels better to me. Or calculating it in `script.js`, and saving the result to `GlobalContext`. What do you think?

Not a big deal either way, and I don't feel strongly.

Reply:
Yeah that's a fair call. I guess I saw the precedent and assumed we were treating it as our derived user state. I agree that mixing data isn't ideal. I'll rethink this 👍

Reply:
I'm personally not too concerned about the derivation, but it since the language is vague, it could be open to future conflicts.

Reply:
Restructured this so that we're leaving the `userRecord` unmodified, apart from the password override. Feels like we should be able to omit the mutation of the password value too, but I looked into it and it's not simple enough to be added to this PR.

Reply:
Doh, I missed that we were changing the password here too, that's a good point. IMO that has the same problem, but you're right that it establishes a precedent and changes the nature of this function.

[6a7290f](https://github.com/WordPress/wporg-two-factor/pull/161/commits/6a7290fd951baa118dbd84ccbf7b15359612204e) LGTM though 👍🏻